packages/wallet-ui/src/components/ui/molecule/AmountInput/AmountInput.view.tsx

```diff
@@ -30,72 +31,64 @@ export const AmountInputView = ({
   error,
   helperText,
   label,
+  value,
   decimalsMax = 18,
   asset,
   onChangeCustom,
   ...otherProps
 }: Props) => {
   const [focused, setFocused] = useState(false);
   const inputRef = useRef<HTMLInputElement>(null);
+  const [inputValue, setInputValue] = useState(value || ''); // Manage the input's value
   const [totalPrice, setTotalPrice] = useState('');
   const [usdMode, setUsdMode] = useState(false);
 
-  const fetchTotalPrice = () => {
-    const inputValue = inputRef.current?.value || '';
+  const fetchTotalPrice = useCallback(() => {
     if (asset.usdPrice && inputValue && inputValue !== '.') {
       const inputFloat = parseFloat(inputValue);
       setTotalPrice(getAmountPrice(asset, inputFloat, usdMode));
     } else {
       setTotalPrice('');
     }
-  };
+  }, [asset, inputValue, usdMode]);
 
-  const resizeInput = () => {
-    if (inputRef.current !== null)
-      inputRef.current.style.width =
-        inputRef.current.value.length * 8 + 6 + 'px';
-  };
+  const resizeInput = useCallback(() => {
+    if (inputRef.current !== null) {
+      inputRef.current.style.width = inputValue.length * 8 + 6 + 'px';
+    }
+  }, [inputValue]);
 
-  const triggerOnChange = () => {
-    //If we are in USD mode we sent the eth amount as the value
-    let valueToSend = inputRef.current?.value || '';
+  const triggerOnChange = (newValue: string) => {
+    setInputValue(newValue);
     if (onChangeCustom) {
-      if (
-        usdMode &&
-        asset.usdPrice &&
-        inputRef.current?.value &&
-        inputRef.current?.value !== '.'
-      ) {
-        const inputFloat = parseFloat(inputRef.current.value);
+      let valueToSend = newValue;
+      if (usdMode && asset.usdPrice && newValue && newValue !== '.') {
+        const inputFloat = parseFloat(newValue);
         valueToSend = getAmountPrice(asset, inputFloat, usdMode);
       }
       onChangeCustom(valueToSend);
     }
   };
 
-  const handleOnKeyUp = () => {
-    //Resize the input depending on content
+  useEffect(() => {
+    // Adjust the input size whenever the value changes
     resizeInput();
-    inputRef.current && fetchTotalPrice();
-  };
-
+  }, [resizeInput]);
```

Comment:
```suggestion
  }, [inputValue]);

```
same comment as below, remove the useCallback